docs/mcd.md

```diff
@@ -0,0 +1,326 @@
+# Modular Connection Dialog (MCD)
```

Comment:
This happens to be what we call it internally.  This is really connection dialog v2 or a name we want to document externally

Reply:
Nice catch. Will use Connection Dialog V2 for now, we can update later.

Reply:
Make sure you also change it below - there's a lot of places you use the acronym MCD that we should change.

Reply:
Replaced all occurrences of "MCD" and "Modular Connection Dialog" with "Tableau Connector" per Stephen's feedback. Except in .png names.